spec/spec.md

```diff
@@ -0,0 +1,133 @@
+
```

Comment:
note: this is moved from the former README.md as-is.

Reply:
I commented above too but I'm thinking we should move the following sections (remove from here and keep at the top level spec/README.md), as they're relevant to all variants, not just the *oci* variant:

- Introduction
- Terminology
- Overview section from within Description

Reply:
OK I bring the common terminology out of spec.md to README.md, and left the spec.md specific ones as-is.